server/etcdserver/api/membership/cluster.go

```diff
@@ -522,8 +522,9 @@ func (c *RaftCluster) PromoteMember(id types.ID, shouldApplyV3 ShouldApplyV3) {
 	defer c.Unlock()
 
 	if c.v2store != nil {
-		if _, ok := c.members[id]; ok {
-			m := *(c.members[id])
+		membersMap, _ := membersFromStore(c.lg, c.v2store)
```

Comment:
```
Start single node cluster
add learner etcd-2, and promote it;
add learner etcd-3, and promote it;
remove etcd-2;
add learner etcd-2 again, and promote it;
reboot any member will trigger the issues:
Error: membership: too many learner members in cluster
Panicking
```

I try to write e2e code to reproduce that issue without your patch. I can't reproduce it right now.
sometimes, if we join member back with same name fast, it will show member is removed.
Just share my testing result.